x-pack/plugins/synthetics/server/rest_api/synthetics_service/add_monitor_push.ts

```diff
@@ -0,0 +1,370 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { isEqual } from 'lodash';
+import { schema } from '@kbn/config-schema';
+import {
+  SavedObjectsUpdateResponse,
+  SavedObjectsClientContract,
+  SavedObjectsFindResult,
+} from '@kbn/core/server';
+import { EncryptedSavedObjectsClient } from '@kbn/encrypted-saved-objects-plugin/server';
+import { UMServerLibs } from '../../lib/lib';
+import {
+  BrowserFields,
+  ConfigKey,
+  MonitorFields,
+  SyntheticsMonitorWithSecrets,
+  EncryptedSyntheticsMonitor,
+  ServiceLocationErrors,
+  PushBrowserMonitor,
+  Locations,
+} from '../../../common/runtime_types';
+import {
+  syntheticsMonitorType,
+  syntheticsMonitor,
+} from '../../lib/saved_objects/synthetics_monitor';
+import { UMRestApiRouteFactory } from '../types';
+import { API_URLS } from '../../../common/constants';
+import { getServiceLocations } from '../../lib/synthetics_service/get_service_locations';
+import { normalizePushedMonitor } from '../../lib/synthetics_service/normalizers/browser';
+import { formatSecrets, normalizeSecrets } from '../../lib/synthetics_service/utils/secrets';
+import { UptimeServerSetup } from '../../lib/adapters/framework';
+import { syncNewMonitor } from './add_monitor';
+import { syncEditedMonitor } from './edit_monitor';
+import { deleteMonitor } from './delete_monitor';
+import { validatePushMonitor } from './monitor_validation';
+
+interface StaleMonitor {
+  stale: boolean;
+  journeyId: string;
+  savedObjectId: string;
+}
+type StaleMonitorMap = Record<string, StaleMonitor>;
+type FailedMonitors = Array<{ id: string; reason: string; details: string; payload?: object }>;
+
+const getSuiteFilter = (projectId: string) => {
+  return `${syntheticsMonitorType}.attributes.${ConfigKey.IS_PUSH_MONITOR}: true AND ${syntheticsMonitorType}.attributes.${ConfigKey.PROJECT_ID}: ${projectId}`;
+};
+
+export const addPublicSyntheticsMonitorRoute: UMRestApiRouteFactory = (libs: UMServerLibs) => ({
+  method: 'PUT',
+  path: API_URLS.SYNTHETICS_MONITORS_PUSH,
+  validate: {
+    body: schema.object({
```

Comment:
Should I be using body validation? Seems like we often don't use it.

Reply:
This would fail if additional fields were sent as well, correct?

Reply:
if we are not doing config-schema body validation, we should use io-ts validation